File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/StudentDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.AccountPrivacy;
import com.example.oriengo.model.enumeration.EducationLevel;
import com.example.oriengo.model.enumeration.VisibilityStatus;
import com.example.oriengo.model.enumeration.MessagePermission;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import com.example.oriengo.model.entity.Location;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class StudentDTO {

    @NotBlank
    private String firstName;
    @NotBlank
    private String lastName;
    @NotBlank
    @Email
    private String email;
    private String phoneNumber;
    private String age;
    private String gender;

    private String school;
    private String fieldOfStudy;
    @NotNull
    private EducationLevel educationLevel;
    private Location location;
    @NotNull
    private VisibilityStatus profileVisibility;
    @NotNull
    private AccountPrivacy accountPrivacy;
    @NotNull
    private MessagePermission messagePermission; // si pertinent


}

File: OrienGO-Frontend/src/app/Service/student.dto.ts
import { LocationDTO } from "../model/dto/Location.dto";
import { AccountPrivacy } from "../model/enum/AccountPrivacy.enum";
import { EducationLevel } from "../model/enum/EducationLevel.enum";
import { GenderType } from "../model/enum/GenderType.enum";
import { MessagePermission } from "../model/enum/MessagePermission.enum";
import { VisibilityStatus } from "../model/enum/VisibilityStatus.enum";


export interface StudentDTO {
  firstName: string;

  lastName: string;

  age: number;

  gender: GenderType;

  phoneNumber?: string;   // optional, since validation allows empty

  email: string;

  password: string;

  // confirmPassword?: string; // uncomment if used in Angular forms

  school?: string;

  fieldOfStudy?: string;

  educationLevel: EducationLevel;

  messagePermission: MessagePermission;

  accountPrivacy: AccountPrivacy;

  profileVisibility: VisibilityStatus;

  location: LocationDTO;

  // Added fields
  enabled: boolean;

  suspended: boolean;

  suspensionReason?: string;

  suspendedUntil?: string;  // LocalDateTime → ISO string
}
